### ISSUE TITLE
Suave (Maybe a)

### ISSUE THREAD

**Issue by tonyday567** [isovector/suavemente#4]
I was trying to implement the comment in #1 

> But there's a trick if you want dynamic elements: code on the javascript side doesn't need to be applicative! It's completely fair game to write an input that responds to other JS events corresponding to its visibility.

First step, for my use case, was to wrap a bunch of inputs in a div, so I could hide it if a checkbox gets unchecked. My bunch of inputs, however, is a `Suave a` thing (composed using the Suave Applicative), and I couldn't work out how to add the div wrapper to the Markup in the Input inside it.  Roughly, I thought of the function I need as `Bool -> Suave a -> Suave (Maybe a)` where Bool is whether the checkbox is checked and `Suave a` is the thing being hidden.

Any thoughts on a path to achieving this?

**Comment by isovector** [comment]
I haven't thought too much about this, but the following seems like a promising avenue. It's all untested.

Add a newtype that has a phantom symbol parameter, corresponding to a group that should all have the same visibility:

```haskell
newtype VisibilityGroup (group :: Symbol) a = VisibilityGroup
  { unVisibilityGroup :: a
  }
type role VisibilityGroup phantom representational 
```

make this thing lift the results of `mkInput`:

```haskell
liftInput :: forall group a. KnownSymbol group => Suave a -> Suave (VisibilityGroup group a)
liftInput (Suave m) = Suave $ do
  let group = symbolVal $ Proxy @group
  Input parser html events <- m
  pure $ Input 
      (coerce parser) 
      (fmap (fmap $ ([qc|<div class="{group}">|] ++) 
                  . (++ "</div>") 
            ) $ coerce html) 
      (coerce events)
```

and now add a new input type whose html injects javacript that hides any div with the class you specified in the visibility group.

**Comment by isovector** [comment]
Fixed by #5
